pkg/apis/serving/v1beta1/predictor.go

```diff
@@ -66,6 +69,14 @@ type PredictorSpec struct {
 	ComponentExtensionSpec `json:",inline"`
 }
 
+type WorkerSpec struct {
+	PodSpec `json:",inline"`
+
+	// Configure the number of replicas in the worker set, each worker set represents the unit of scaling
+	// +optional
+	Size int `json:"size,omitempty"`
```

Comment:
Shall we consider using the pipeline parallelism terminology which is more familiar to ML engineers?

Reply:
ML engineers still use the pipeline parallell size.

- If the pipeline-parallel-size is set in the environment variables, that value will be utilized.
- If the value isn't set and the size is specified instead, the size will be assigned the value of pipeline-parallel-size.
-  If both are not specified, the default value will be assigned the value(2 -> 1 head/1worker) of pipeline-parallel-size.

Now, both options will be available for configuration.

Reply:
how about rename `size` to `pipeline-parallel-size` ? any benefits need both field and the environment variables?  it is also easier to validate fields for pipeline parallelism

Reply:
I find that `pipelineParallelismSize` can be somewhat confusing in the context of WorkerSpec.

**Is there a need for both the field and the environment variables?**

- For MLOps engineers, specifying size in WorkerSpec seems like a straightforward approach to increasing the number of worker nodes.
- On the other hand, Data Scientists might find it more intuitive to use pipeline-parallel-size.

>it is also easier to validate fields for pipeline parallelism

In the end, the validating part is to check whether the `worker node size` and `pipeline-parellel-size` are the same or different. As I explained above, the size is determined according to the priority of each field, and this value is set to `worker node size` and `pipeline-parellel-size`, so I think the validating part will be resolved.

Reply:
I find the naming closely related to a chicken-egg problem:
* By naming it `size` our naming will be aligned to LeaderWorkerSet.
* By naming it `pipelineParallelismSize`, it is going to be more familiar for people using vLLM.

I, personally, like more the `size` naming because it is more related to infrastructure config (which is closer to KServe concern) than to the technique used for distributing the load.

Reply:
I am fine with `size` as long as we document it well. Consistency with LWS is a good consideration since LWS is more battle-tested through API and use case discussions.

Reply:
pipeline parallelism is not just vLLM concept, it is a common LLM inference terminology and kserve is in a good position to standardize these concepts instead of making them second citizen fields.
https://developer.nvidia.com/blog/mastering-llm-techniques-inference-optimization/

Reply:
I am not sure if following LWS is a strong argument.

Reply:
Sorry about the strong opinion, but I really want to make kserve focusing on LLM inference specification and standardize these concepts for GenAI.

Reply:
Discussed with @Jooho , will make a proposed change in a separate PR, good to approve this one.

Reply:
> Sorry about the strong opinion, but I really want to make kserve focusing on LLM inference specification and standardize these concepts for GenAI.

Just to contribute to the discussion... Well, the way I've understood the `ServingRuntime` CRD is that it is a generic CRD for configuring any runtime you'd like to use. This was my motivation for preferring the more generic `size` naming. Under this understanding, using `pipelineParallelismSize` didn't make sense to me, as we anyway would need the runtime to support it and, AFAIK, KServe would still be limited to passing down this value to the runtime container (e.g. through envvars), and it's going to be the runtime the one applying the config (or ignoring it).

Now... I'm not against using `pipelineParallelismSize` naming, but if we do it my thought is that KServe would be communicating a closer integration with the model servers, as not any server implements pipeline parallelism. Maybe is it better to introduce a specialized CRD for LLM inferencing? We just discussed pipeline paralellism size, but we could easily add to this discussion the tensor paralellism size that also would be important (e.g. 2 GPUs on each node).